src/scicat_metadata.py

```diff
@@ -140,20 +141,30 @@ def from_file(cls, schema_file_name: pathlib.Path) -> "MetadataSchema":
         return cls.from_dict(_load_json_schema(schema_file_name))
 
 
-def render_variable_value(var_value: str, variable_registry: dict) -> str:
+def render_variable_value(var_value: Any, variable_registry: dict) -> str:
+    # if input is not a string it converts it to string
+    output_value = var_value if isinstance(var_value, str) else json.dumps(var_value)
+
     # If it is only one variable, then it is a simple replacement
-    if (var_key := var_value.removesuffix(">").removeprefix("<")) in variable_registry:
+    if (
+        var_key := output_value.removesuffix(">").removeprefix("<")
+    ) in variable_registry:
         return variable_registry[var_key]
 
     # If it is a complex variable, then it is a combination of variables
     # similar to f-string in python
     for reg_var_name, reg_var_value in variable_registry.items():
-        var_value = var_value.replace("<" + reg_var_name + ">", str(reg_var_value))
+        output_value = output_value.replace(
+            "<" + reg_var_name + ">", str(reg_var_value)
+        )
 
     if "<" in var_value and ">" in var_value:
         raise Exception(f"Unresolved variable: {var_value}")
 
-    return var_value
+    output_value = (
+        output_value if isinstance(var_value, str) else json.loads(output_value)
+    )
```

Comment:
I'm not sure about the intention here. Can you elaborate more? @nitrosx

Reply:
In some cases, the variable value (aka var_value) is a list or a dict.
When that is the case, the variable value is converted to a json string, the substitution is executed and converted back to the original type.
Basically it is a trick to perform nested substitutions without traversing the whole nested structure.

Check also line 149.

Reply:
Ah it's making the string dumped `output_value` back to a dictionary/list. I see... 
It sounds like a bit of hack but it won't be a problem since `>` and `<` are not easy to be mistaken with other symbols used in list or dictionary but we'd better document it.
I'll do it in another PR with some unit tests.